torch/csrc/utils/tensor_new.cpp

```diff
@@ -204,10 +204,32 @@ void recursive_store(char* data, IntArrayRef sizes, IntArrayRef strides, int64_t
   for(const auto i : c10::irange(n)) {
 #ifdef USE_NUMPY
     if (is_numpy_available() && PyArray_Check(items[i])) {
-      TORCH_WARN_ONCE(
-        "Creating a tensor from a list of numpy.ndarrays is extremely slow. "
-        "Please consider converting the list to a single numpy.ndarray with "
-        "numpy.array() before converting to a tensor.");
+      auto np_arr = (PyArrayObject*)items[i];
```

Comment:
Is this guaranteed to be a NumPy array? Do we have to deal with cases like a list of lists of lists? Is our testing of this functionality very thorough?